Article/Article.js

```diff
@@ -1,26 +1,131 @@
-// Because classes are not hoised you will need to start your code at the bottom of the page.  Look for the comment "START HERE"
+// Because classes are not hoised you will need to 
+//start your code at the bottom of the page. 
+// Look for the comment "START HERE"
 
 class Article {
   constructor(domElement) {
     // assign this.domElement to the passed in domElement
-    this.domElement;
+    this.domElement = domElement;
     // create a reference to the ".expandButton" class. 
-    this.expandButton;
-    // Using your expandButton reference, update the text on your expandButton to say "expand"
-    
-    // Set a click handler on the expandButton reference, calling the expandArticle method.
-    
+    this.expandButton = domElement.querySelector('.expandButton');
+    // Using your expandButton reference, update the text on your 
+    //expandButton to say "expand"
+    this.expandButton.innerText = 'EXPAND';
+
+    // Set a click handler on the expandButton reference, 
+    //calling the expandArticle method.
+    this.expandButton.addEventListener('click', this.expandArticle.bind(this));
+
   }
 
   expandArticle() {
-    // Using our reference to the domElement, toggle a class to expand or hide the article.
-
+    // Using our reference to the domElement, 
+    //toggle a class to expand or hide the article.
+    this.domElement.classList.toggle('article-open');
+    if (this.domElement.classList.contains('article-open')) {
+      this.expandButton.innerText = "CLOSE";
+    }
+    else if (!this.domElement.classList.contains('article-open')) {
+      this.expandButton.innerText = "EXPAND";
+    }
   }
 }
 
-// START HERE: Select all classes named ".article" and assign that value to the articles variable
-let articles;
+// START HERE: Select all classes named ".article" and 
+//assign that value to the articles variable
+let articles = document.querySelectorAll('.article');
+
+// Use .map() to iterate over the articles array and 
+//create a new instance of Article by passing in 
+//each article as a parameter to the constructor.
+
+articles = Array.from(articles).map(article => new Article(article));
+
+//    FORM BOX EXPERIMENT  \\
+
+let formBox = document.querySelector('.form-box');
+
+formBox.addEventListener('click', function () {
+  formBox.classList.toggle('reg-form');
+  formBox.classList.remove('btn-form');
+}, { once: true })
+
+let publishBtn = document.querySelector('.publish-btn');
+
+let articleContainer = document.querySelector('.articles');
+
+
+
+
+let titleInput = document.querySelector('#title').value;
+let dateInput = null;
+let paraInput = document.querySelector('textarea').value;
 
-// Use .map() to iterate over the articles array and create a new instance of Article by passing in each article as a parameter to the constructor.
-articles = Array.from(articles);
 
+
+
+class Articles {
```

Comment:
it looks like this creates a single Article right? So probably don't pluralize the name?